builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -195,7 +195,7 @@ func resourceArmVirtualMachine() *schema.Resource {
 
 						"disk_size_gb": {
 							Type:         schema.TypeInt,
-							Required:     true,
+							Optional:     true,
```

Comment:
What happens if the value is not supplied on creation?

`expandAzureRmVirtualMachineDataDisk` seems to assume the value is always set.